Make MVCC setting a no-op

katzyn opened the pull request:
A second part of changes for #1204.

`MVCC` setting is now ignored. MVCC logic is always used for MVStore and never used for PageStore.

I removed a couple of tests for table-level locking with disabled MVCC in MVStore mode, this is not possible with MVCC. I also changed a one of PageStore tests to use auto-commit feature, because this test used MVCC and without MVCC it can only get a lock timeout.

`Database.isMultiVersion()` is renamed to `isMVStore()`.

katzyn committed:
Remove MVCC setting and use MVCC logic if and only if MVStore is used

katzyn committed:
Update advanced.html

katzyn commented:
Failure on Travis is not related.

Elimination of dead code was intentionally left for a future pull request(s).

@grandinj, @andreitokar
Could you review it?

grandinj commented on h2/src/main/org/h2/table/MetaTable.java:
```diff
@@ -1043,7 +1043,7 @@ private boolean hideTable(Table table, Session session) {
             add(rows, "MODE", database.getMode().getName());
             add(rows, "MULTI_THREADED", database.isMultiThreaded() ? "1" : "0");
-            add(rows, "MVCC", database.isMultiVersion() ? "TRUE" : "FALSE");
+            add(rows, "MVCC", database.isMVStore() ? "TRUE" : "FALSE");
```
fairly sure we can drop this line of code

grandinj commented:
other than that one comment, LGTM

katzyn committed:
Remove MVCC from INFORMATION_SCHEMA.SETTINGS

katzyn commented:
OK, I removed it. There is also a JMX method, but its removal will be an API-breaking change. May be it's better not to remove it right now.

GC overhead limit exceeded in `TestOptimizations` with MVStore on Java 7 in one of “usual” failures on Travis.

Modified tests that used non-default MVCC setting are completed normally. Other tests should not be affected in any may.
